server/src/main/java/org/opensearch/node/InternalSettingsPreparer.java

```diff
@@ -81,6 +81,11 @@ public static Environment prepareEnvironment(Settings input, Map<String, String>
         initializeSettings(output, input, properties);
         Environment environment = new Environment(output.build(), configPath);
 
+        if (Files.exists(environment.configFile().resolve("elasticsearch.yml"))) {
+            throw new SettingsException(
+                "elasticsearch.yml was deprecated with opensearch and must be renamed to opensearch.yml");
+        }
+
         if (Files.exists(environment.configFile().resolve("opensearch.yaml"))) {
             throw new SettingsException("opensearch.yaml was deprecated in 5.5.0 and must be renamed to opensearch.yml");
         }
```

Comment:
It's not in scope of this PR but I think this can be removed or the message can modified to say only the extension `.yml` is supported.

Reply:
Will take this as part of https://github.com/opensearch-project/OpenSearch/issues/358 and remove the corresponding tests.